owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/security/PassCodeViewModel.kt

```diff
@@ -0,0 +1,59 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author Juan Carlos Garrote Gascón
+ *
+ * Copyright (C) 2021 ownCloud GmbH.
+ * <p>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.presentation.viewmodels.security
+
+import androidx.lifecycle.ViewModel
+import com.owncloud.android.data.preferences.datasources.SharedPreferencesProvider
+import com.owncloud.android.presentation.ui.security.PassCodeActivity
+
+class PassCodeViewModel(
+    private val preferencesProvider: SharedPreferencesProvider
+) : ViewModel() {
+
+    fun setPassCode(passcode: String) {
+        preferencesProvider.putString(PassCodeActivity.PREFERENCE_PASSCODE, passcode)
+        preferencesProvider.putBoolean(PassCodeActivity.PREFERENCE_SET_PASSCODE, true)
+    }
+
+    fun removePassCode() {
+        preferencesProvider.putBoolean(PassCodeActivity.PREFERENCE_SET_PASSCODE, false)
```

Comment:
Don't we remove the current passcode? I mean, we keep the `PREFERENCE_PASSCODE` stored after removing the passcode, but should we delete it? 🤔

Reply:
In the current version it is not removed. We could delete it, but anyways the passcode is only read when `PREFERENCE_SET_PASSCODE` is true, and when it is put to true again (the moment when you register a new passcode) `PREFERENCE_PASSCODE` is overwritten, so there shouldn't be problem (the app has been working like this until now). Maybe removing it or setting it to null is safer, what do you think?

Reply:
I think that removing it would be the best option 🤔